lua/dap/ui/variable-tree.lua

```diff
@@ -0,0 +1,246 @@
+local api = vim.api
+local utils = require("dap.utils")
+local ui = require("dap.ui.share")
+local non_empty_sequence = utils.non_empty_sequence
+
+local M = {}
+
+M.multiline_variable_display = false
+M.max_win_width = 100
+M.default_sidebar_width = 50
+M.variable_value_separator = " = "
+
+local floating_buf = nil
+local floating_win = nil
+local variable_buffers = {}
+
+local function open_bufwin(split_cmd, winopts, steal_focus)
+  local prev_win = api.nvim_get_current_win()
+  vim.cmd(split_cmd)
+  local win = api.nvim_get_current_win()
+  ui.apply_winopts(win, winopts)
+
+  local buf = api.nvim_create_buf(false, true)
+  api.nvim_win_set_buf(win, buf)
+
+  if steal_focus ~= false then
+    api.nvim_set_current_win(prev_win)
+  end
+  return buf, win
+end
+
+local function open_sidebar(steal_focus, winopts)
+  winopts = winopts or {width = M.default_sidebar_width}
+  return open_bufwin("vsplit", winopts, steal_focus)
+end
+
+local function update_variable_buffer(buf, win)
+  api.nvim_buf_set_option(buf, "modifiable", true)
+  local state = variable_buffers[buf]
+  M.write_variables(buf, state.variables, 0, 0, win)
+end
+
+function M.toggle_variable_expanded()
+  local buf = api.nvim_get_current_buf()
+  local state = variable_buffers[buf]
+  local pos = api.nvim_win_get_cursor(0)
+  local win = api.nvim_get_current_win()
+
+  local v = state.line_to_variable[pos[1] - 1]
```

Comment:
It should not be possible. The autocmd and the state are created in successive line. It could happen if the user calls this internal function. I added a check.